common/static/js/src/tooltip_manager.js

```diff
@@ -1,109 +1,239 @@
 (function() {
     'use strict';
-    var TooltipManager = function (element) {
-        this.element = $(element);
-        // If tooltip container already exist, use it.
-        this.tooltip = $('div.' + this.className.split(/\s+/).join('.'));
-        // Otherwise, create new one.
-        if (!this.tooltip.length) {
-            this.tooltip = $('<div />', {
-                'class': this.className
-            }).appendTo(this.element);
-        }
-
-        this.hide();
-        _.bindAll(this, 'show', 'hide', 'showTooltip', 'moveTooltip', 'hideTooltip', 'click');
-        this.bindEvents();
-    };
 
-    TooltipManager.prototype = {
-        // Space separated list of class names for the tooltip container.
+    var Tooltip = Backbone.View.extend({
         className: 'tooltip',
-        SELECTOR: '[data-tooltip]',
 
-        bindEvents: function () {
-            this.element.on({
-                'mouseover.TooltipManager': this.showTooltip,
-                'mousemove.TooltipManager': this.moveTooltip,
-                'mouseout.TooltipManager': this.hideTooltip,
-                'click.TooltipManager': this.click
-            }, this.SELECTOR);
+        template: _.template('<div class="tooltip-content"><%= content %></div><div class="tooltip-arrow">▾</div>'),
```

Comment:
CSS pseudo-elements can't be styled or positioned, and we need to be able to move the arrow to the side if the tooltip position is offset by the edge of the viewport.